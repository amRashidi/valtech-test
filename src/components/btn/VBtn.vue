<script setup lang="ts">
import { computed } from "@vue/runtime-core"
import { LocationAsRelativeRaw } from "vue-router"

    const props =defineProps<{router?:boolean;rounded?:boolean;to?:LocationAsRelativeRaw; outline?:boolean,text?:boolean;dark?:boolean;disabled?:boolean}>()
    const btnComponent = computed(() => !!props.router ? 'router-link': 'button')
    const btnAttrs = computed(() => ({
        disabled: props.disabled,
        to: props.to
    }))
    const btnClasess = computed(() => ({
        "v-btn": true,
        "v-btn--outlined": props.outline,
        "v-btn--disabled": props.disabled,
        "v-btn--rounded": props.rounded,
        "v-btn--text": props.text,
        "v-btn--dark": props.dark,
    }))
    const componentRole = computed(() => !!props.router ? "link" : "button")
</script>

<template>
    <component :is="btnComponent" v-bind="btnAttrs" :class="btnClasess" tabindex="0" :role="componentRole">
        <div class="v-btn__content">
            <slot />
        </div>
    </component>
</template>

<style lang="scss" scoped>
@import "@/assets/styles/btn";
</style>