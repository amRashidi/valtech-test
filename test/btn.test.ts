import { mount } from '@vue/test-utils';
import VBtn from "../src/components/btn/VBtn.vue";

test('mount btn', () => {
    expect(VBtn).toBeTruthy()
    const btn = mount(VBtn)
    expect(btn.classes()).toContain("v-btn");
})
test('btn slot', () => {
    const wrapper = mount(VBtn, {
        slots: {
            default: 'Read More',
        }
    })

    expect(wrapper.text()).toContain('Read More')
})
test('btn classses', async () => {
    expect(VBtn).toBeTruthy()
    const btn = mount(VBtn)
    expect(btn.classes()).toEqual(["v-btn"])
    await btn.setProps({ dark: true, text: true, disabled: true, outline: true })
    expect(btn.classes()).toEqual(["v-btn", "v-btn--outlined", "v-btn--disabled", "v-btn--text", "v-btn--dark"])
})
test('btn as router component', async () => {
    expect(VBtn).toBeTruthy()
    const btn = mount(VBtn, {
        props: {
        }
    })
    expect(btn.classes()).toEqual(["v-btn"])
    await btn.setProps({ router: true })
    expect(btn.find('router-link').exists()).toBe(true)
})