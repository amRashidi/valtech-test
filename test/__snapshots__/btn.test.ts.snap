// Vitest Snapshot v1

exports[`btn slot 1`] = `
"<button class=\\"v-btn\\" data-v-4983db4e=\\"\\">
  <div class=\\"v-btn__content\\" data-v-4983db4e=\\"\\">Read More</div>
</button>"
`;

exports[`component btn 1`] = `
"<router-link disabled=\\"false\\" class=\\"v-btn\\" data-v-4983db4e=\\"\\">
  <div class=\\"v-btn__content\\" data-v-4983db4e=\\"\\"></div>
</router-link>"
`;
